plugins/modules/site_workflow_manager.py

```diff
@@ -555,111 +554,147 @@ def get_current_site(self, site):
                                 if isinstance(site, dict):
                                     current_site = dict(site.items())
-                                    if site.get('nameHierarchy'):
-                                        current_site['parentName'] = site['nameHierarchy'].rsplit('/', 1)[0]
-                                    else:
-                                        current_site['parentName'] = None
-
+                                    current_site['parentName'] = site.get('nameHierarchy', '').rsplit('/', 1)[0] if site.get('nameHierarchy') else None
                                     all_sites_info.append(current_site)
-                                    site_exists = True
+
+                            site_exists = True
+
                         except Exception as e:
                             self.log("Error fetching site for name '{0}': {1}".format(name, str(e)))
 
-                    return (site_exists, all_sites_info)
+                    if all_sites_info:
+                        self.log("All site information collected from bulk operation: {}".format(all_sites_info), "DEBUG")
+                        return site_exists, current_site
 
-            except Exception as e:
-                self.log("Bulk operation is not available due to : {0}".format(str(e)), "WARNING")
-                name_hierarchy = self.want.get("site_name")
-                try:
-                    response = self.get_site(name_hierarchy)
-                except Exception as e:
-                    self.log("Error fetching site for {0}: {1}".format(name_hierarchy, str(e)))
-                response = response.get("response")
-                if not response:
-                    self.log("No site information found for name hierarchy: {0}".format(name_hierarchy), "WARNING")
-                    return (site_exists, current_site)
+                name_hierarchy = self.want.get("site_name_hierarchy")
+                response = self.get_site(name_hierarchy)
 
-                if response:
-                    self.log("Received API response from 'get_sites': {0}".format(
-                        str(response)), "DEBUG")
-                    all_sites_info = []
-                    for site in response:
-                        if isinstance(site, dict):
-                            current_site = dict(site.items())
-                            name_hierarchy = site.get('nameHierarchy', '')
-                            if name_hierarchy:
-                                current_site['parentName'] = name_hierarchy.rsplit('/', 1)[0]
-                            else:
-                                current_site['parentName'] = None
+                if not response:
+                    self.log("No site information found for name hierarchy: {}".format(name_hierarchy), "WARNING")
+                    return site_exists, current_site
+
+                response_data = response.get("response", [])
```

Comment:
We can change response_data into sites.. so we can use in for loop, will be more readable

`for site in sites:`

Reply:
Modified to sites